src/stores/user-store.js

```diff
@@ -57,6 +57,15 @@ class UserStore extends EventEmitter {
     getUsers () {
         return this._users;
     }
+
+    checkUserNameExist (user) {
```

Comment:
This way can be possibly used.
But our target is to be use `actions` and `events` as much as possible.
I offer you to rewite the logic with following approach:
1) component `User` check fields validity (as you already do)
2) call store's action `addNewUser`
3) store catches this action and check what it should check
4) store emit an event eith the result of the action handling
- user added
- we have an error and all required information about error in args
5) the component get this event and handle it